api/rosetta/validator.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package rosetta
+
+import (
+	"errors"
+
+	"github.com/optakt/flow-dps/rosetta/identifier"
+)
+
+type Validator interface {
+	Request(interface{}) error
+	CompleteBlockID(identifier.Block) error
+}
+
+var (
+	ErrBlockLength     = errors.New(BlockLength)
+	ErrBlockIncomplete = errors.New(BlockNotFull)
+	ErrAddressLength   = errors.New(AddressLength)
+	ErrTxLength        = errors.New(TxLength)
+)
```

Comment:
I would prefer to replace these with error types, rather than sentinel errors. That way, you can use `errors.As` instead of `errors.Is` in the error handling and keep all of the information about the errors in the Rosetta error.